lib/x11utils.pm

```diff
@@ -106,8 +106,14 @@ sub ensure_unlocked_desktop {
             record_info('Guest disabled display', 'Might be consequence of bsc#1168979');
         }
         if (match_has_tag('authentication-required-user-settings') || match_has_tag('authentication-required-modify-system')) {
-            type_password;
-            assert_and_click "authenticate";
+            wait_still_screen(2);
+            if (!check_screen([qw(authentication-required-user-settings authentication-required-modify-system)])) {
```

Comment:
nitpicking (optional): it is easy to read if you reverse the if/else and avoid '!', so:
1. wait still screen
2. if authentication still there, type password and assert
3. else record soft-failure and early exit with next